mllib/src/test/scala/org/apache/spark/ml/linalg/VectorUDTSuite.scala

```diff
@@ -44,4 +45,39 @@ class VectorUDTSuite extends SparkFunSuite {
     assert(dataType.asInstanceOf[StructType].fields.map(_.dataType)
       === Seq(new VectorUDT, DoubleType))
   }
+
+  test("SPARK-28158 Hive UDFs supports UDT type") {
```

Comment:
We can put this test in the sql/hive module, and create a UDT for testing.

Reply:
ok